src/main/java/org/sagebionetworks/bridge/config/PropertiesConfig.java

```diff
@@ -0,0 +1,243 @@
+     *
+     * @param delimiterRegex
+     *            The regular expression for the delimiter that is used to
+     *            separate a list of values.
+     */
+    public PropertiesConfig(final Path configTemplate, final Path localConfig,
+            final String delimiterRegex) throws IOException {
+        checkNotNull(configTemplate);
+        checkNotNull(localConfig);
+        checkNotNull(delimiterRegex);
+        final Properties properties = new Properties();
+        try (final Reader templateReader = Files.newBufferedReader(configTemplate);
+                final Reader localReader = Files.newBufferedReader(localConfig)) {
+            properties.load(templateReader);
+            properties.load(localReader);
+        }
+        user = readUser(properties);
+        environment = readEnvironment(properties);
+        final Properties collapsed = collapse(properties, environment.name().toLowerCase());
+        this.properties = new Properties(collapsed);
+        delimiter = Pattern.compile(delimiterRegex);
+    }
+
+    @Override
+    public String getUser() {
+        return user;
+    }
+
+    @Override
+    public Environment getEnvironment() {
+        return environment;
+    }
+
+    @Override
+    public String get(final String key) {
+        checkNotNull(key);
+        return properties.getProperty(key);
+    }
+
+    @Override
+    public int getInt(final String key) {
+        checkNotNull(key);
+        return Integer.parseInt(properties.getProperty(key));
+    }
+
+    @Override
+    public List<String> getList(final String key) {
+        checkNotNull(key);
+        final String property = properties.getProperty(key);
+        return Lists.newArrayList(delimiter.split(property));
+    }
+
+    private final ConfigReader envReader = new ConfigReader() {
+        @Override
+        public String read(String name) {
+            try {
+                // Change to a valid environment variable name
+                name = name.toUpperCase().replace('.', '_');
+                return System.getenv(name);
+            } catch (SecurityException e) {
+                throw new GetEnvException(name, e);
+            }
+        }
+    };
+
+    private final ConfigReader sysReader = new ConfigReader() {
+        @Override
+        public String read(String name) {
+            try {
+                return System.getProperty(name);
+            } catch (SecurityException e) {
+                throw new GetSystemPropertyException(name, e);
+            }
+        }
+    };
+
+    private String readUser(final Properties properties) {
+        String user = read(USER_KEY, properties);
+        if (Strings.isNullOrEmpty(user)) {
+            return DEFAULT_USER;
+        }
+        return user;
+    }
+
+    private Environment readEnvironment(final Properties properties) {
+        final String envName = read(ENV_KEY, properties);
+        if (envName == null) {
+            return DEFAULT_ENV;
+        }
```

Comment:
Minor: Should blank envName also use the default env? (This would be consistent with user.)